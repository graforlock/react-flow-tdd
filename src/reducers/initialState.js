// @flow
import type { EditorStore } from './editorReducer'

type ReduxState = {
  ideas: Array<Object>,
  editor: EditorStore,
  sortOption: ?string
}

const initialState: ReduxState = {
  ideas: [],
  editor: {
    writingMode: false,
    editedIdea: null
  },
  sortOption: null
}

export default initialState
export type { ReduxState }
