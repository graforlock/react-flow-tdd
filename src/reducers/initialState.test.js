// @flow
import initialState from './initialState'
import type { ReduxState } from './initialState'

describe('initialState', () => {
  let state: ReduxState

  beforeEach(() => {
    state = initialState
  })

  it('matches the current initialState snapshot', () => {
    expect(state).toMatchSnapshot()
  })
})
