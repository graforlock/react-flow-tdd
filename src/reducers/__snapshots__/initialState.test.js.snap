// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialState matches the current initialState snapshot 1`] = `
Object {
  "editor": Object {
    "editedIdea": null,
    "writingMode": false,
  },
  "ideas": Array [],
  "sortOption": null,
}
`;
